src/CacheValue.ts

```diff
@@ -0,0 +1,152 @@
+import { Either } from 'fp-ts/lib/Either';
+import { Functor2 } from 'fp-ts/lib/Functor';
+import { Setoid, fromEquals } from 'fp-ts/lib/Setoid';
+import { constFalse, constant } from 'fp-ts/lib/function';
+
+declare module 'fp-ts/lib/HKT' {
+  interface URI2HKT2<L, A> {
+    CacheValue: CacheValue<L, A>;
+  }
+}
+
+export const URI = 'CacheValue';
+
+export type URI = typeof URI;
+
+export type CacheValue<L, A> =
+  | CacheValueInitial<L, A>
```

Comment:
why did we keep the Initial? If we really want it, why didn't we use RemoteData?

Reply:
Note that this is not the ADT exposed to components (`QueryValue`), it is the one used internally by the cache. The initial case is actually used when a cache for a new input is initialised (even if soon afterwards it will be a "Pending", since they are only initialised when actually requested by someone), if I remember correctly I added it to have better types in the implementation of `ObservableCache.ts`

> why didn't we use RemoteData?

I think because we are carrying around different data (we have `value` and `updated`)